src/qibo/config.py

```diff
@@ -41,23 +41,39 @@ def raise_error(exception, message=None, args=None):
     # Backend access
     K = tf
 
-    # Set the intra number of threads to the environment flag
-    if "QIBO_NUM_THREADS" in os.environ: # pragma: no cover
-        nthreads = int(os.environ["QIBO_NUM_THREADS"])
-        tf.config.threading.set_intra_op_parallelism_threads(nthreads)
+    # Set the number of threads from the environment variable
+    OMP_NUM_THREADS = None
+    if "OMP_NUM_THREADS" not in os.environ:
+        import psutil
+        # using physical cores by default
+        cores = psutil.cpu_count(logical=False)
```

Comment:
Here this is the number of physical cores which is typically half of the total threads that appear in `htop`. This should be okay since in the benchmarks above we see that performance saturates beyond this number. If we remove `logical=False` we get the maximum number of threads.

Reply:
Yes, indeed, I am using the physical cores (which should similar to what tf does with inter/intra = 0).
Concerning this line, probably tensorflow could tell use the number of physical cores, but I didn't mange to find the right function, so I am using psutils (and unfortunately adding its dependency).